### PR TITLE
feat(wallet) Implement new design for NFT details screen(wallet v2)

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/30286

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

https://github.com/brave/brave-core/assets/48117473/65965669-ea8c-445b-9a9b-3c2c046be711

### CODE REVIEW

File: components/resources/wallet_strings.grdp [hunk 1]
```diff
@@ -514,7 +514,17 @@
...
+  <message name="IDS_BRAVE_WALLET_NFT_DETAIL_IMAGE_ADDRESS" desc="NFT detail image address category">Image URL</message>
+  <message name="IDS_BRAVE_WALLET_NFT_DETAILS_PINNING_IN_PROGRESS" desc="NFT detail pinning in progress">In progress</message>
+  <message name="IDS_BRAVE_WALLET_NFT_DETAILS_PINNING_SUCCESSFUL" desc="NFT detail pinning successful">Pinned</message>
+  <message name="IDS_BRAVE_WALLET_NFT_DETAILS_PINNING_FAILED" desc="NFT detail pinning failed">Failed</message>
+  <message name="IDS_BRAVE_WALLET_NFT_DETAILS_OVERVIEW" desc="NFT detail overview">Overview</message>
+  <message name="IDS_BRAVE_WALLET_NFT_DETAILS_OWNED_BY" desc="NFT detail owned by">Owned by</message>
+  <message name="IDS_BRAVE_WALLET_NFT_DETAILS_VIEW_ACCOUNT" desc="NFT detail view account">View account</message>
+  <message name="IDS_BRAVE_WALLET_NFT_DETAILS_PINNING_STATUS_LABEL" desc="NFT detail pinning status label">IPFS pinning status</message>
+  <message name="IDS_BRAVE_WALLET_NFT_DETAILS_NOT_AVAILABLE" desc="NFT detail not avaialable text">Not available yet</message>
+  <message name="IDS_BRAVE_WALLET_NFT_DETAILS_CID" desc="NFT detail CID label">CID</message>
```
Comment (mkarolin) [inline components/resources/wallet_strings.grdp:526]: This one is probably `translateable="false"`?

Comment (muliswilliam) [reply components/resources/wallet_strings.grdp:526]: @mkarolin Please clarify what that means

Comment (mkarolin) [reply components/resources/wallet_strings.grdp:526]: If this string doesn't need to be localized to other languages, then we should add `translateable="false"` attribute to the `<message`.

Comment (mkarolin) [reply components/resources/wallet_strings.grdp:526]: Just for `IDS_BRAVE_WALLET_NFT_DETAILS_CID`

Comment (muliswilliam) [reply components/resources/wallet_strings.grdp]: @mkarolin Fixed 5f31d7291a6863fbccf6d1322c4a95997b811b5e

Comment (mkarolin) [reply components/resources/wallet_strings.grdp]: Thanks!

**Review by mkarolin — APPROVED** [review]
`strings`++

File: components/brave_wallet_ui/nft/components/nft-details/nft-screen.tsx [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+    }
+    if (selectedAsset && isNftPinnable) {
+      return getNftPinningStatus(selectedAsset)
+    }
+    return undefined
+  }, [isNftPinnable, selectedAsset, nftPinningStatus, isAutoPinEnabled])
+
+  const isNftPinned = React.useMemo(() => {
+    return currentNftPinningStatus?.code === BraveWallet.TokenPinStatusCode.STATUS_PINNED
+  }, [currentNftPinningStatus])
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/nft/components/nft-details/nft-screen.tsx:137]:
```suggestion
  }, [currentNftPinningStatus?.code])
```

File: components/brave_wallet_ui/nft/components/nft-details/nft-screen.tsx [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+        case BraveWallet.TokenPinStatusCode.STATUS_PINNED:
+        return getLocale('braveWalletNFTDetailsPinningSuccessful')
+
+        case BraveWallet.TokenPinStatusCode.STATUS_PINNING_FAILED:
+        return getLocale('braveWalletNFTDetailsPinningFailed')
+   
+      default:
+        return getLocale('braveWalletNFTDetailsNotAvailable')
+    }
+  }, [currentNftPinningStatus])
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/nft/components/nft-details/nft-screen.tsx:159]:
```suggestion
  }, [currentNftPinningStatus?.code])
```

File: components/brave_wallet_ui/nft/components/nft-details/nft-screen.tsx [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+
+  const tokenId = React.useMemo(
+    () =>
+      selectedAsset.tokenId ? new Amount(selectedAsset.tokenId).toNumber() : '',
+    [selectedAsset]
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/nft/components/nft-details/nft-screen.tsx:164]:
```suggestion
    [selectedAsset.tokenId]
```

File: components/brave_wallet_ui/components/desktop/card-headers/nft-asset-header.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+          <ButtonIcon
+            size={16}
+            name='arrow-left'
+          />
+        </CircleButton>
+        <HeaderTitle>
+          {assetName}&nbsp;{tokenId ? `#${new Amount(tokenId).toNumber()}` : ''}
+        </HeaderTitle>
+      </Row>
+      <SendButton>{getLocale('braveWalletSend')}</SendButton>
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/card-headers/nft-asset-header.tsx:55]: Should this `SendButton` route to `brave://wallet/send`?

Comment (muliswilliam) [reply components/brave_wallet_ui/components/desktop/card-headers/nft-asset-header.tsx]: Fixed here: 8cec20024b47506f0b3efb3edf2b2fa09aadc62e

**Review by darkdh — APPROVED** [review]
unblocked l10n string constants

**Pull request merged**